core.coffee

```diff
@@ -390,14 +443,22 @@
 
 			get: (params) ->
 				singular = utils.isObject(params) and params.id?
-				return @request(params, method: 'GET').then (data) ->
-					if !data?.d?
-						throw new Error('Invalid response received.')
-					if singular
-						if data.d.length > 1
-							throw new Error('Returned multiple results when only one was expected.')
-						return data.d[0]
-					return data.d
+
+				requestFn = =>
+					return @request(params, method: 'GET').then (data) ->
```

Comment:
You should be able to do
```coffee
compiledUrl = @compile(params)
if utils.isString(params)
	params = compiledUrl 
else
	params.url = compiledUrl
```
once outside of this function and then `@request` won't have to recompile for every poll